tests/unit/test_irc.py

```diff
@@ -351,12 +333,13 @@ def test_configure_evaluates_configuration_file_local(irc_bridge_service, mocker
         mocker.call(f"{IRC_BRIDGE_CONFIG_FILE_PATH.absolute()}", "w", encoding="utf-8"),
     ]
 
-    mock_open.assert_has_calls(calls, any_order=True)
+    mock_builtin_open.assert_has_calls(calls, any_order=True)
     mock_safe_load.assert_called_once_with(
-        mock_open().__enter__()  # pylint: disable=unnecessary-dunder-call
+        mock_builtin_open().__enter__()  # pylint: disable=unnecessary-dunder-call
     )
     mock_dump.assert_called_once_with(
-        mock_safe_load(), mock_open().__enter__()  # pylint: disable=unnecessary-dunder-call
+        mock_safe_load(),
```

Comment:
Having a quick look at the test, it tests that certain functions are called (open file, load...). Maybe a better option is to test the output of a function with its inputs (and the output is a file). pytest has fixtures like tmpdir and monkeypatch that can be useful.

Reply:
I'd leave that as an additional option in the future, the tests are rather basic atm. I agree with you.